sky/serve/service.py

```diff
@@ -116,15 +116,17 @@ def _cleanup(service_name: str) -> bool:
             logger.error(f'Replica {info.replica_id} failed to terminate.')
     versions = serve_state.get_service_versions(service_name)
     serve_state.remove_service_versions(service_name)
-    success = True
-    for version in versions:
+
+    def cleanup_version_storage(version: int) -> bool:
         task_yaml: str = serve_utils.generate_task_yaml_file_name(
             service_name, version)
         logger.info(f'Cleaning up storage for version {version}, '
                     f'task_yaml: {task_yaml}')
-        success = success and cleanup_storage(task_yaml)
-    if not success:
+        return cleanup_storage(task_yaml)
```

Comment:
This cleanup storage should only be execute once

Reply:
What do you mean by “execute once”. IIUC, the previous code cleans up storage for each versions, until encounters a failure with a version.

Reply:
oh nvm, i misread the code. please ignore this one